### PR TITLE
Cloud client library migration part 1: Registry management

### PR DESCRIPTION
**Pull request by gguuss**

### CODE REVIEW

**Comment by gguuss** [comment]
FYI the tests are failing outside of this change.

File: iot/manager/manager.js [hunk 1]
```diff
@@ -120,34 +126,42 @@ const createRegistry = async (
...
-  // getClient(serviceAccountJson, function(client) {...});
   // const cloudRegion = 'us-central1';
   // const projectId = 'adjective-noun-123';
   // const registryId = 'my-registry';
   // function errCb = lookupRegistry; // Lookup registry if already exists.
-  const parentName = `projects/${projectId}/locations/${cloudRegion}`;
-  const pubsubTopic = `projects/${projectId}/topics/${pubsubTopicId}`;
+  const iot = require('@google-cloud/iot');
 
+  // Lookup the pubsub topc
```
Comment (hongalex) [inline iot/manager/manager.js]: nit: s/topc/topic

File: iot/manager/manager.js [hunk 2]
```diff
@@ -120,34 +126,42 @@ const createRegistry = async (
...
   try {
-    const {data} = await client.projects.locations.registries.create(request);
+    const responses = await iotClient.createDeviceRegistry(request);
+    const response = responses[0];
 
     console.log('Successfully created registry');
-    console.log(data);
+    console.log(response);
   } catch (err) {
     console.log('Could not create registry');
```
Comment (hongalex) [inline iot/manager/manager.js]: Can we make this (and other error logging) something like:

`console.error('Could not create registry: ', err)`

File: iot/manager/manager.js [hunk 3]
```diff
@@ -577,24 +594,30 @@ const clearRegistry = async (client, registryId, projectId, cloudRegion) => {
...
+  );
+  try {
+    const responses = await iotClient.deleteDeviceRegistry({
+      name: registryName,
+    });
+    console.log(responses);
     console.log('Successfully deleted registry');
-    console.log(res);
   } catch (err) {
     console.log('Could not delete registry');
```
Comment (hongalex) [inline iot/manager/manager.js]: Collapse error into one line:

`console.log('Could not delete registry: ', err);`

File: iot/manager/manager.js [hunk 4]
```diff
@@ -799,17 +822,27 @@ const getRegistry = async (client, registryId, projectId, cloudRegion) => {
...
+      registryId
+    );
+    try {
+      const responses = await iotClient.getDeviceRegistry({name: registryName});
+      const response = responses[0];
+
+      console.log('Found registry:', registryId);
+      console.log(response);
+    } catch (err) {
+      console.error(err);
```
Comment (hongalex) [inline iot/manager/manager.js]: Add error context here

**Comment by gguuss** [comment]
@hongalex PTAL, updated the logging as appropriate and addressed nits.

File: iot/manager/manager.js [hunk 5]
```diff
@@ -120,34 +126,41 @@ const createRegistry = async (
...
+    const responses = await iotClient.createDeviceRegistry(request);
+    const response = responses[0];
 
     console.log('Successfully created registry');
-    console.log(data);
+    console.log(response);
   } catch (err) {
-    console.log('Could not create registry');
-    console.log(err);
+    console.log('Could not create registry', err);
```
Comment (hongalex) [inline iot/manager/manager.js]: Actually, can we print to `stderr` for all of these changes?

`console.error('Could not create registry', err);` is probably better.

File: iot/manager/manager.js [hunk 6]
```diff
@@ -197,8 +210,7 @@ const createDevice = async (
     console.log('Created device');
     console.log(data);
   } catch (err) {
-    console.log('Could not create device');
-    console.log(err);
+    console.log('Could not create device', err);
```
Comment (hongalex) [inline iot/manager/manager.js]: use `console.error`

File: iot/manager/manager.js [hunk 7]
```diff
@@ -236,8 +248,7 @@ const createUnauthDevice = async (
...
```
Comment (hongalex) [inline iot/manager/manager.js]: use `console.error`

File: iot/manager/manager.js [hunk 8]
```diff
@@ -287,8 +298,7 @@ const createRsaDevice = async (
...
```
Comment (hongalex) [inline iot/manager/manager.js]: use `console.error`

File: iot/manager/manager.js [hunk 9]
```diff
@@ -336,8 +346,7 @@ const createEsDevice = async (
...
```
Comment (hongalex) [inline iot/manager/manager.js]: use `console.error`

File: iot/manager/manager.js [hunk 10]
```diff
@@ -383,8 +392,7 @@ const patchRsa256ForAuth = async (
     console.log('Patched device:', deviceId);
     console.log(data);
   } catch (err) {
-    console.log('Error patching device:', deviceId);
-    console.log(err);
+    console.log('Error patching device:', deviceId, err);
```
Comment (hongalex) [inline iot/manager/manager.js]: use `console.error`

File: iot/manager/manager.js [hunk 11]
```diff
@@ -430,8 +438,7 @@ const patchEs256ForAuth = async (
...
```
Comment (hongalex) [inline iot/manager/manager.js]: use `console.error`

File: iot/manager/manager.js [hunk 12]
```diff
@@ -457,31 +464,33 @@ const listDevices = async (client, registryId, projectId, cloudRegion) => {
     );
     console.log('Current devices in registry:', data['devices']);
   } catch (err) {
-    console.log('Could not list devices');
-    console.log(err);
+    console.log('Could not list devices', err);
```
Comment (hongalex) [inline iot/manager/manager.js]: use `console.error`

File: iot/manager/manager.js [hunk 13]
```diff
@@ -514,8 +523,7 @@ const deleteDevice = async (
     console.log('Successfully deleted device:', deviceId);
     console.log(data);
   } catch (err) {
-    console.log('Could not delete device:', deviceId);
-    console.log(err);
+    console.log('Could not delete device:', deviceId, err);
```
Comment (hongalex) [inline iot/manager/manager.js]: use `console.error`

File: iot/manager/manager.js [hunk 14]
```diff
@@ -577,24 +585,30 @@ const clearRegistry = async (client, registryId, projectId, cloudRegion) => {
...
+    const responses = await iotClient.deleteDeviceRegistry({
+      name: registryName,
+    });
+    console.log(responses);
     console.log('Successfully deleted registry');
-    console.log(res);
   } catch (err) {
     console.log('Could not delete registry');
-    console.log(err);
+    console.error(err);
```
Comment (hongalex) [inline iot/manager/manager.js]:
```
console.error('Could not delete registry:', err);
```

File: iot/manager/manager.js [hunk 15]
```diff
@@ -799,17 +813,27 @@ const getRegistry = async (client, registryId, projectId, cloudRegion) => {
...
+      registryId
+    );
+    try {
+      const responses = await iotClient.getDeviceRegistry({name: registryName});
+      const response = responses[0];
+
+      console.log('Found registry:', registryId);
+      console.log(response);
+    } catch (err) {
+      console.error(err);
```
Comment (hongalex) [inline iot/manager/manager.js]:
```
console.error('Could not get device registry:',err);
```

File: iot/manager/manager.js [hunk 16]
```diff
@@ -891,50 +923,54 @@ const setIamPolicy = async (
...
+    bindings.forEach(_binding => {
+      console.log(`Role: ${_binding.role}`);
+      _binding.members || (_binding.members = []);
+      _binding.members.forEach(_member => {
+        console.log(`\t${_member}`);
+      });
+    });
   } catch (err) {
     console.log('Could not set policy for: ', registryId);
     console.log('Trace: ', err);
```
Comment (hongalex) [inline iot/manager/manager.js]: Replace both of these with `console.error`

**Comment by gguuss** [comment]
I went ahead and cleaned up the whole file, there was inconsistent use of console error and log.

**Pull request merged**